### PR TITLE
Implemented Factory reset

### PR DESCRIPTION
**Pull request by SteffiAntony**
jira: https://jsw.ibm.com/browse/PFEBMC-2382

### CODE REVIEW

File: src/views/Operations/FactoryReset/FactoryResetModal.vue [hunk 1]
```diff
@@ -1,23 +1,26 @@
 <template>
-  <b-modal
+  <BModal
     id="modal-reset"
     ref="modal"
+    v-model="modal"
```
Comment (nabilpandiyalayil) [inline src/views/Operations/FactoryReset/FactoryResetModal.vue]: This is no longer required.

File: src/views/Operations/FactoryReset/FactoryReset.vue [hunk 2]
```diff
@@ -1,140 +1,136 @@
...
+<script setup>
+import { ref, onMounted, computed, defineEmits } from 'vue';
+import Alert from '../../../components/Global/Alert.vue';
+import PageTitle from '../../../components/Global/PageTitle.vue';
+import { GlobalStore } from '@/store/modules/GlobalStore';
+import ModalReset from './FactoryResetModal.vue';
+import useLoadingBar from '@/components/Composables/useLoadingBarComposable';
+import useToastComposable from '@/components/Composables/useToastComposable';
+import AuthenticationStore from '../../../store/modules/Authentication/AuthenticationStore';
+import FactoryResetStore from '@/store/modules/Operations/FactoryResetStore';
```
Comment (nabilpandiyalayil) [inline src/views/Operations/FactoryReset/FactoryReset.vue]: Import these from '@/store'

File: src/views/Operations/FactoryReset/FactoryReset.vue [hunk 3]
```diff
@@ -1,140 +1,136 @@
...
-import Alert from '@/components/Global/Alert';
-import PageTitle from '@/components/Global/PageTitle';
-import BVToastMixin from '@/components/Mixins/BVToastMixin';
-import LoadingBarMixin from '@/components/Mixins/LoadingBarMixin';
-import ModalReset from './FactoryResetModal';
+<script setup>
+import { ref, onMounted, computed, defineEmits } from 'vue';
+import Alert from '../../../components/Global/Alert.vue';
+import PageTitle from '../../../components/Global/PageTitle.vue';
+import { GlobalStore } from '@/store/modules/GlobalStore';
```
Comment (nabilpandiyalayil) [inline src/views/Operations/FactoryReset/FactoryReset.vue]: Just import from '@/store'

File: src/views/Operations/FactoryReset/FactoryReset.vue [hunk 4]
```diff
@@ -1,140 +1,136 @@
...
-<script>
-import Alert from '@/components/Global/Alert';
-import PageTitle from '@/components/Global/PageTitle';
-import BVToastMixin from '@/components/Mixins/BVToastMixin';
-import LoadingBarMixin from '@/components/Mixins/LoadingBarMixin';
-import ModalReset from './FactoryResetModal';
+<script setup>
+import { ref, onMounted, computed, defineEmits } from 'vue';
+import Alert from '../../../components/Global/Alert.vue';
+import PageTitle from '../../../components/Global/PageTitle.vue';
```
Comment (nabilpandiyalayil) [inline src/views/Operations/FactoryReset/FactoryReset.vue]: Use '@' to import

File: src/views/Operations/FactoryReset/FactoryReset.vue [hunk 5]
```diff
@@ -1,140 +1,136 @@
...
+const { hideLoader, startLoader, endLoader } = useLoadingBar();
+const resetOption = ref('resetBios');
+const serverStatus = computed(() => {
+  return Global.serverStatus;
+});
+onMounted(() => {
+  hideLoader();
+  emit('loading-bar-status', true);
+});
+const emit = defineEmits(['loading-bar-status']);
```
Comment (nabilpandiyalayil) [inline src/views/Operations/FactoryReset/FactoryReset.vue]: Please explain the need for these emits

**Review by nabilpandiyalayil — CHANGES_REQUESTED** [review]

File: src/views/Operations/FactoryReset/FactoryReset.vue (hunk 5 above)
Comment (SteffiAntony) [reply src/views/Operations/FactoryReset/FactoryReset.vue]: Sorry. Just forgot to remove these lines

**Review by Nikhil-Ashoka — CHANGES_REQUESTED** [review]
The styling is not as it was in Factory Reset page in Vue2 code.
The text should be properly aligned as it is in vue2 and Button must have gap.

**Comment by Nikhil-Ashoka** [comment]
Screenshot 2024-08-23 at 1 02 01 PM

**Comment by Nikhil-Ashoka** [comment]
Screenshot 2024-08-23 at 8 31 03 PM

File: src/store/modules/Operations/FactoryResetStore.js [hunk 6]
```diff
@@ -10,25 +11,27 @@ const FactoryResetStore = {
...
     async resetBios() {
       return await api
         .post('/redfish/v1/Systems/system/Bios/Actions/Bios.ResetBios')
-        .then(() => i18n.t('pageFactoryReset.toast.resetBiosSuccess'))
+        .then(() => i18n.global.t('pageFactoryReset.toast.resetBiosSuccess'))
         .catch((error) => {
           console.log('Factory Reset: ', error);
-          throw new Error(i18n.t('pageFactoryReset.toast.resetBiosError'));
+          throw new Error(
+            i18n.globalt('pageFactoryReset.toast.resetBiosError'),
```
Comment (nabilpandiyalayil) [inline src/store/modules/Operations/FactoryResetStore.js]: fix the typo "globalt"

File: src/store/modules/Settings/PowerPolicyStore.js
Comment (nabilpandiyalayil) [inline src/store/modules/Settings/PowerPolicyStore.js:1]: why this file in this PR?

File: src/views/Settings/PowerRestorePolicy/PowerRestorePolicy.vue
Comment (nabilpandiyalayil) [inline src/views/Settings/PowerRestorePolicy/PowerRestorePolicy.vue:1]: Here also

**Review by nabilpandiyalayil — CHANGES_REQUESTED** [review]

File: src/store/modules/Settings/PowerPolicyStore.js
Comment (SteffiAntony) [reply src/store/modules/Settings/PowerPolicyStore.js:1]: By resolving a merge conflict after taking the latest pull, this file was pushed.

File: src/views/Settings/PowerRestorePolicy/PowerRestorePolicy.vue
Comment (SteffiAntony) [reply src/views/Settings/PowerRestorePolicy/PowerRestorePolicy.vue:1]: By resolving a merge conflict after taking the latest pull, this file was pushed.

File: src/router/routes.js [hunk 7]
```diff
@@ -88,6 +89,23 @@ export const routes = [
...
+        },
+      },
+      {
+        path: '/settings/power-restore-policy',
+        name: 'power-restore-policy',
+        component: PowerRestorePolicy,
+        meta: {
+          title: i18n.global.t('appPageTitle.powerRestorePolicy'),
+        },
+      },
```
Comment (nabilpandiyalayil) [inline src/router/routes.js]: This should not be added again. Please pull the latest changes.

File: src/views/Operations/FactoryReset/FactoryReset.vue [hunk 8]
```diff
@@ -1,140 +1,142 @@
...
+import { ref, onMounted, computed } from 'vue';
+import Alert from '@/components/Global/Alert.vue';
+import PageTitle from '@/components/Global/PageTitle.vue';
+import ModalReset from './FactoryResetModal.vue';
+import useLoadingBar from '@/components/Composables/useLoadingBarComposable';
+import useToastComposable from '@/components/Composables/useToastComposable';
+import { FactoryResetStore, AuthenticationStore, GlobalStore } from '@/store';
+import eventBus from '@/eventBus';
+
+const Global = GlobalStore();
```
Comment (Nikhil-Ashoka) [inline src/views/Operations/FactoryReset/FactoryReset.vue]: let the variable names start with lowercase: "globalStore", "authenticationStore", "factoryResetStore", "toast"

**Review by Nikhil-Ashoka — CHANGES_REQUESTED** [review]

**Review by Nikhil-Ashoka — APPROVED** [review]

**Review by nabilpandiyalayil — APPROVED** [review]

**Pull request merged**